modules/reitit-middleware/src/reitit/ring/middleware/session.clj

```diff
@@ -0,0 +1,33 @@
+(ns reitit.ring.middleware.session
+  (:require
+   [clojure.spec.alpha :as s]
+   [ring.middleware.session :as session]
+   [ring.middleware.session.memory :as memory]))
+
+(s/def ::spec (s/keys :opt-un [::session]))
+
+(def ^:private store
+  "The default shared in-memory session store.
+
+  This is used when no `:session` key is provided to the middleware."
+  (atom {}))
+
+(def session-middleware
+  "Middleware for session.
+
+  Enter:
+  Add the `:session` key into the request map based on the `:cookies`
+  in the request map.
+
+  Exit:
+  When `:session` key presents in the response map, update the session
+  store with its value. Then remove `:session` from the response map.
+
+  | key          | description |
+  | -------------|-------------|
+  | `:session`   | `ring.middleware.session.store/SessionStore` instance. Use `ring.middleware.session.memory/MemoryStore` by default."
```

Comment:
I think it would be good to use the same options as the underlaying mw: https://github.com/ring-clojure/ring/blob/master/ring-core/src/ring/middleware/session.clj#L85-L100

Reply:
Thanks! I'm on it.